lib/ui/ui_dart_state.cc

```diff
@@ -187,6 +189,13 @@ void UIDartState::ReportUnhandledException(const std::string& error,
                  << stack_trace;
 }
 
+void UIDartState::LogMessage(const std::string& tag,
+                             const std::string& message) const {
+  if (log_message_callback_) {
```

Comment:
For existing embedders that don't provide a callback, this would mean their logs are no longer visible. This is a breaking change. IMO, there should be platform specific defaults that embedders may override if necessary.

Reply:
I've put workarounds in each of the existing embedders that don't use the embedder API. There's a fallback in the embedder API as well, so this shouldn't be a breaking change.

Reply:
Done.